src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/inmemoryrepo/InMemoryCourseRepositoryTest.kt

```diff
@@ -9,65 +9,63 @@ import io.kotest.matchers.nulls.shouldNotBeNull
 import io.kotest.matchers.should
 import io.kotest.matchers.shouldBe
 import org.junit.jupiter.api.Test
-import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.CourseService
-import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.inmemoryrepo.InMemoryCourseRepository
-import java.util.UUID
+import java.util.*
 
-class CourseServiceTest {
-  private val service = CourseService(InMemoryCourseRepository())
+class InMemoryCourseRepositoryTest {
```

Comment:
In the previous commit `CourseServiceTest` tests an instance of `CourseService`. (The CourseService instance has a CourseRepository instance injected in turn). This demonstrates that the refactoring hasn't broken the CourseService tests. That is, I'm changing code without changing tests.

In the commit above I then re-purpose CourseServiceTest to test InMemoryCourseRepository directly.  So in this commit I'm changing the test, not the code. The CourseService instance is removed from the test and replaced by the repository.  That is why I renamed the test class in this commit because it is testing a different class.

For this reason I think that moving the rename to the previous commit would be wrong.

You could argue that the second commit (the rename commit) should be split in two  - 1) rename and repurpose the tests, 2) Add flyway and flyway migration for CourseEntity.